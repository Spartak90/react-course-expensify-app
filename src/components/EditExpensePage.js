import React from 'react';
import { connect } from 'react-redux';
import { editExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class EditExpensePage extends React.Component {
  onSubmit = () => {
    this.props.editExpense(this.props.match.params.id, this.props.expense);
    this.props.history.push('/');
  }

  render () {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit} />
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    expense: state.expenses.find((expense) => {
      return expense.id === ownProps.match.params.id;
    })
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    editExpense: (id, expense) => dispatch(editExpense(id, expense))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
