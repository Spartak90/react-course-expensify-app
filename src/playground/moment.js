import moment from 'moment';

const date = moment(0);
const dateBefore = moment(0).subtract(4, 'days').valueOf();
const dateAfter = moment(0).add(4, 'days').valueOf();

console.log(
    'isBefore', date, moment(dateBefore), date.isSameOrBefore(moment(dateBefore))
);

console.log(
    'isAfter', date, moment(dateAfter), date.isSameOrAfter(moment(dateAfter))
);
