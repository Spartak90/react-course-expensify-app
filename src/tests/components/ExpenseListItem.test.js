import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseListItem } from '../../components/ExpenseListItem';
import moment from 'moment';

const expense = {
    id: 1,
    description:
    'Water bill',
    amount: 4500,
    createdAt: moment(0)
};

test('should render ExpenseListItem with expenses', () => {
    const wrapper = shallow(<ExpenseListItem {...expense} />);

    expect(wrapper).toMatchSnapshot();
});
