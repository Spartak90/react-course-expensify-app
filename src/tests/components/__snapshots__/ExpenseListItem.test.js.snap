// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with expenses 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Water bill
    </h3>
  </Link>
  <p>
    4500
     - 
    <Component />
  </p>
</div>
`;
